address.md

```diff
@@ -0,0 +1,300 @@
+3. **Upgradable**: Addresses must be versioned to permit the introduction of new address formats.
+4. **Compact**: Given the above constraints, addresses must be as short as possible.
+
+## Specification
+
+A Filecoin address encoded to a byte array consists of:
+* A **protocol indicator** byte that identifies the type and version of this address.
+* The **payload** used to uniquely identify the actor according to the protocol.
+```
+|----------|---------|
+| protocol | payload |
+|----------|---------|
+|  1 byte  | n bytes |
+```
+
+A Filecoin address encoded to a string consistes of:
+* A **network prefix** character that identifies the network the address belongs to.
+* A **protocol indicator** byte that identifies the type and version of this address.
+* A **payload** used to uniquely identify the actor according to the protocol..
+* A **checksum** used to validet the address. 
+
+```
+|------------|----------|---------|----------|
+|  network   | protocol | payload | checksum |
+|------------|----------|---------|----------|
+| 'f' or 't' |  1 byte  | n bytes | 4 bytes  |
+```
+
+An example of a Filecoin address in golang:
+
+```go
+type Address struct {
+    
+    // 0: ID
+    // 1: Blake2b160-Hash of secp256k1 Public Key
+    // 2: Blake2b160-Hash of Actor creation data
+    // 3: BLS Public Key
+    Protocol byte
+    
+    // raw bytes containing the data associated with protocol
+    Payload []byte
+}
+```
+
+#### Network Prefix
+
+The **network prefix** is prepended to an address when encoding to a string. The network prefix indicates which network an address belongs in. The network prefix may either be `f` for filecoin mainnet or `t` for filecoin testnet. Is it worth noting that a network prefix will never appear on chain and is only used when encoding an address to a human readable format.
+
+#### Protocol Indicator
+
+The **protocol indicator** byte describes how a method should interpret the information in the payload field of an address. Any deviation for the algorithms and data types specified by the protocol must be assigned a new protocol number. In this way, protocols also act as versions.
+
+- `0`: ID
+- `1` : SECP256K1 Public Key
+- `2` : Actor
+- `3` : BLS Public Key
+
+An example description in golang:
+
+```go
+// Protocol byte
+type Protocol = byte
+const (
+    ID Protocol = iota
+    SECP256K1
+    Actor
+    BLS
+)
+```
+
+###### Protocol 0: IDs
+
+**Protocol 0** addresses are simple IDs. These addresses are expected to remain within the network. All actors have a numeric ID even if they don't have public keys. The payload of an ID address is encoded as decimal numbers '0'-'9'. IDs are not hashed and do not have a checksum.
+
+**Bytes**
+
+```
+|----------|---------|
+| protocol | payload |
+|----------|---------|
+|    0     | '0'-'9' |
+```
+
+**String**
+
+```
+|------------|----------|---------|
+|  network   | protocol | payload |
+|------------|----------|---------|
+| 'f' or 't' |     0    | '0'-'9' |
```

Comment:
probably put '0' in quotes to indicate we're putting it in as a string literal

Reply:
also this is a little misleading. Reading this, I would expect the bytes for the payload to be the same for the raw and stringed versions, when its not. We should clarify that in the stringed version its base10 encoded